### PR TITLE
PORTALS-2624: refactor ModalDownload to use MUI components

### PR DESCRIPTION
**Pull request by hallieswan**
step | main | feature
:---:|:---:|:---:
1 | main_ModalDownload_1 | feature_ModalDownload_1
2 | main_ModalDownload_2 | feature_ModalDownload_2

### CODE REVIEW

File: packages/synapse-react-client/src/lib/containers/ModalDownload.FormSchema.ts [hunk 1]
```diff
@@ -9,7 +9,6 @@ export const tsvOption = 'Tab Separated Values (TSV)'
 
 // Step 1
 const stepOneFormSchema: JSONSchema7 = {
-  title: 'Download query results',
```
Comment (hallieswan) [inline packages/synapse-react-client/src/lib/containers/ModalDownload.FormSchema.ts:12]: Move title to the Dialog, so the styling matches other MUI dialogs

File: packages/synapse-react-client/src/lib/containers/ModalDownload.tsx [hunk 2]
```diff
@@ -126,59 +126,59 @@ export default class ModalDownload extends React.Component<
...
-                  <div style={spinnerStyle} className="spinner" />
-                  <p> Loading... </p>
+      <DialogBase
+        open={true}
+        onCancel={this.props.onClose}
+        title="Download query results"
+        content={
+          <>
+            <Form
+              id="modal-download-form"
```
Comment (hallieswan) [inline packages/synapse-react-client/src/lib/containers/ModalDownload.tsx:142]: Add a form id so the buttons can live in the actions section of the dialog (outside of the form), but still submit the form on click

File: packages/synapse-react-client/src/lib/containers/ModalDownload.tsx [hunk 3]
```diff
@@ -126,59 +126,59 @@ export default class ModalDownload extends React.Component<
...
-              </Button>
-              <Button type="submit" variant="primary">
-                {this.state.step === 0 ? 'Next' : 'Download'}
-              </Button>
-            </div>
-          </Form>
-        </Modal.Body>
-      </Modal>
+              )}
+              <React.Fragment />
```
Comment (hallieswan) [inline packages/synapse-react-client/src/lib/containers/ModalDownload.tsx:158]: Prevents the [submit button](https://github.com/rjsf-team/react-jsonschema-form/issues/1602#issuecomment-648614139) from appearing in the Form

File: packages/synapse-react-client/src/lib/containers/ModalDownload.tsx [hunk 4]
```diff
@@ -126,59 +126,59 @@ export default class ModalDownload extends React.Component<
...
+          <>
+            <Button
+              variant="outlined"
+              color="primary"
+              onClick={this.props.onClose}
+            >
+              Cancel
+            </Button>
+            <Button
+              form="modal-download-form"
```
Comment (hallieswan) [inline packages/synapse-react-client/src/lib/containers/ModalDownload.tsx:172]: Links [this button](https://github.com/mui/material-ui/issues/6128#issuecomment-351883972) to the form

Comment (nickgros) [reply packages/synapse-react-client/src/lib/containers/ModalDownload.tsx:172]: Nice!

**Review by nickgros — APPROVED** [review]

**Pull request merged**